### PR TITLE
Multi stage docker build

### PR DESCRIPTION
**Pull request by owl352**
# Issue
Currently, everything is included in the resulting container, thus there are really poor pushing / pulling images sizes and times. We need to split building of the container from the running environment

# Things done
* Added multi stage build in indexer `Dockerfile`
* Updated `build.yml` 
* Created github repo with `Dockerfile` and auto push to docker hub https://github.com/owl352/refinery_cli_docker
* Created package in docker hub with `refinery_cli` https://hub.docker.com/r/owl352/refinery_cli_docker

### CODE REVIEW

File: packages/indexer/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-
-RUN cargo build
+RUN rustup target add x86_64-unknown-linux-musl
+RUN cargo build --target x86_64-unknown-linux-musl --release
 RUN cargo install refinery_cli
+
+FROM scratch as indexer
+WORKDIR /app
+COPY --from=build /app/target/x86_64-unknown-linux-musl/release /app  
+CMD ["/app/indexer"]
```
Comment (pshenmic) [inline packages/indexer/Dockerfile:17]: If you are changing the way image command executes, you should also reflect that in the CI deploy workflow

File: packages/indexer/Dockerfile [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 COPY Cargo.toml /app
 COPY migrations /app/migrations
 COPY src /app/src
-
-RUN cargo build
+RUN rustup target add x86_64-unknown-linux-musl
+RUN cargo build --target x86_64-unknown-linux-musl --release
 RUN cargo install refinery_cli
+
+FROM scratch as indexer
```
Comment (pshenmic) [inline packages/indexer/Dockerfile:14]: Why scratch? I see we use alpine in the build section, maybe do here it too? I'm afraid scratch can have potential issues if indexer would use some feature from a static library which is not provided in the scratch by default. I think it's better to just use alpine

File: packages/indexer/Dockerfile [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 RUN git clone --depth 1 --branch v1.0.0-beta.4  https://github.com/dashevo/platform
-
 WORKDIR /app
 COPY Cargo.lock /app
 COPY Cargo.toml /app
 COPY migrations /app/migrations
 COPY src /app/src
-
-RUN cargo build
+RUN rustup target add x86_64-unknown-linux-musl
```
Comment (pshenmic) [inline packages/indexer/Dockerfile:10]: Is it necessary? Will it work for arm64?

**Review by pshenmic — CHANGES_REQUESTED** [review]

File: packages/indexer/Dockerfile (hunk 1 above)
Comment (owl352) [reply packages/indexer/Dockerfile]: ✅

File: packages/indexer/Dockerfile (hunk 2 above)
Comment (owl352) [reply packages/indexer/Dockerfile]: ✅

**Review by pshenmic — APPROVED** [review]

**Pull request merged**